### PR TITLE
fix: issue with displaying product price od PDP and PLP

### PR DESCRIPTION
**Pull request by bartoszherba**
## Description
- fix the issue with displaying prices of PDP
- fix the issue with displaying prices of PLP
- fix calculation of prices for the bundle, grouped products

## Related Issue
#973 

## Screenshots (if appropriate):
Screenshot 2022-05-24 at 10 11 42

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/modules/catalog/category/components/views/CategoryProductPrice.vue [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+export default defineComponent({
+  name: 'CategoryProductPrice',
+  components: {
+    SfPrice,
+  },
+  props: {
+    product: {
+      type: Object as PropType<typeof useProductsWithCommonProductCardProps | null>,
+      default: () => null,
```
Comment (sethidden) [inline packages/theme/modules/catalog/category/components/views/CategoryProductPrice.vue:67]: Shouldn't this be required? You're accessing product.__typename in the template which will crash if `product` is null

**Review by sethidden — DISMISSED** [review]

File: packages/theme/modules/catalog/category/components/views/CategoryProductPrice.vue [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+} from '~/modules/catalog/category/components/views/useProductsWithCommonCardProps';
+
+export default defineComponent({
+  name: 'CategoryProductPrice',
+  components: {
+    SfPrice,
+  },
+  props: {
+    product: {
+      type: Object as PropType<typeof useProductsWithCommonProductCardProps | null>,
```
Comment (sethidden) [inline packages/theme/modules/catalog/category/components/views/CategoryProductPrice.vue:66]:
```suggestion
            type: Object as PropType<ReturnType<typeof useProductsWithCommonProductCardProps>['productsWithCommonProductCardProps']['value'][number] | null>,
```
Unfortunately this is the type you're looking for. Right now in this PR it's "the product prop is a function or null (typeof useProduct... is a function that accepts products)

What I suggested is the modified product returned by the commoncardprops

File: packages/theme/modules/catalog/category/components/views/CategoryProductPrice.vue (hunk 1 above)
Comment (bartoszherba) [reply packages/theme/modules/catalog/category/components/views/CategoryProductPrice.vue:67]: fixed

File: packages/theme/modules/catalog/category/components/views/CategoryProductPrice.vue (hunk 2 above)
Comment (bartoszherba) [reply packages/theme/modules/catalog/category/components/views/CategoryProductPrice.vue:66]: Awful...

**Review by sethidden — APPROVED** [review]

**Pull request merged**